apps/condo/domains/acquiring/utils/billingCentrifuge.js

```diff
@@ -0,0 +1,343 @@
+            const overpaymentShare = isLast ? restUndistributedOverpaymentAmount : Big(d.overpaymentPart).div(totalOverpaymentReceiversParts).times(totalOverpaymentAmount)
+            const roundedOverpaymentShare = overpaymentShare.round(decimalPlaces, 1)
+            restUndistributedOverpaymentAmount = isLast ? Big(0) : restUndistributedOverpaymentAmount.minus(roundedOverpaymentShare)
+            if (splitIndex >= 0) {
+                const newAmount = Big(splits[splitIndex].amount || 0).plus(roundedOverpaymentShare)
+                splits[splitIndex].amount = newAmount.toString()
+            } else {
+                splits.push({
+                    recipient: d.recipient,
+                    amount: roundedOverpaymentShare.toString(),
+                })
+            }
+        }
+    }
+
+    //
+    // Extract fee from splits
+    //
+    let restUndistributedFeeAmount = Big(feeAmount)
+    if (restUndistributedFeeAmount.gt(0)) {
+        let feePayersTotalSharesAmount = Big(0)
+        let hasSplitsWithFeePayers = false
+        for (let i = 0; i < splits.length; i++) {
+            const split = splits[i]
+            const key = createRecipientKey(split.recipient)
+            if (distributionsByKey[key].isFeePayer) {
+                feePayersTotalSharesAmount = feePayersTotalSharesAmount.plus(split.amount || 0)
+                hasSplitsWithFeePayers = true
+            }
+        }
+        if (hasSplitsWithFeePayers) {
+            for (let i = 0; i < splits.length; i++) {
+                const recipientKey = createRecipientKey(splits[i].recipient)
+                const d = distributionsByKey[recipientKey]
+                if (!d.isFeePayer) {
+                    continue
+                }
+
+                const isLast = i + 1 >= splits.length
+                const feeShare = isLast ? restUndistributedFeeAmount : Big(splits[i].amount).div(feePayersTotalSharesAmount).times(totalFeeAmount)
+                const roundedFeeShare = feeShare.round(decimalPlaces, 1)
+                restUndistributedFeeAmount = isLast ? Big(0) : restUndistributedFeeAmount.minus(roundedFeeShare)
+                splits[i].feeAmount = roundedFeeShare.toString()
+                const newAmount = Big(splits[i].amount).minus(roundedFeeShare)
+                if (newAmount.lt(0)) {
+                    throw new Error(`Recipient ${JSON.stringify(splits[i].recipient)} has amount=${splits[i].amount} and feeAmount=${splits[i].feeAmount}`)
+                }
+                splits[i].amount = newAmount.toString()
+            }
+        } else {
+            // There are no receivers who must pay fee, but feeAmount was set.
+            // So we decrease origin sum and distribute without fee
+            // Than add extracted fee to separated split without recipient
+            const splitsWithoutFee = split(Big(paymentAmount).minus(feeAmount).toString(), distribution, {
+                ...options,
+                feeAmount: undefined,
```

Comment:
why not `0`

Reply:
I've omitted the `feeAmount` field from the original options